In [2]:
import pandas as pd
df = pd.read_csv("C:/TechLabs/PandasApp/myenv_1/gym_membership_prediction_with_professional_status.csv")
df

User ID  Age  Gender  Annual Income Membership Type  \
0           1   62   Other         135305         Premium   
1           2   65    Male          59056           Basic   
2           3   18    Male          63857           Basic   
3           4   21    Male         135072           Basic   
4           5   21  Female          59344           Basic   
...       ...  ...     ...            ...             ...   
3995     3996   40    Male         127601         Premium   
3996     3997   40  Female          77944         Premium   
3997     3998   62    Male          33154           Basic   
3998     3999   55  Female         139822         Premium   
3999     4000   52  Female          61604           Basic   

      Membership Duration  Frequency of Visit Preferred Time  \
0                      46                   2      Afternoon   
1                      26                   6      Afternoon   
2                      46                   2      Afternoon   
3                      54                   1        Evening   
4                      36                   5        Morning   
...                   ...                 ...            ...   
3995                   15                   1      Afternoon   
3996                   34                   1        Evening   
3997                   40                   1        Evening   
3998                   39                   3      Afternoon   
3999                    6                   6        Morning   

     Previous Membership Marital Status  Location Promotion Response  \
0                    Yes        Married     Rural                Yes   
1                    Yes         Single  Suburban                Yes   
2                     No        Married     Rural                Yes   
3                     No        Married  Suburban                 No   
4                    Yes         Single  Suburban                 No   
...                  ...            ...       ...                ...   
3995                 Yes         Single     Urban                 No   
3996                  No         Single     Urban                 No   
3997                 Yes        Married  Suburban                 No   
3998                 Yes         Single     Urban                 No   
3999                 Yes         Single  Suburban                Yes   

     Health Condition Membership Prediction Professional   Professional Status  
0             Average                    No  Late Career               Retired  
1                Good                    No      Retired               Retired  
2           Excellent                    No      Student               Student  
3                Good                   Yes      Student               Student  
4           Excellent                   Yes      Student               Student  
...               ...                   ...          ...                   ...  
3995             Good                    No   Mid Career  Working Professional  
3996          Average                    No   Mid Career  Working Professional  
3997             Good                   Yes  Late Career               Retired  
3998             Poor                   Yes  Late Career  Working Professional  
3999          Average                   Yes  Late Career  Working Professional  

[4000 rows x 16 columns]

In [4]:
df.info


<bound method DataFrame.info of       User ID  Age  Gender  Annual Income Membership Type  \
0           1   62   Other         135305         Premium   
1           2   65    Male          59056           Basic   
2           3   18    Male          63857           Basic   
3           4   21    Male         135072           Basic   
4           5   21  Female          59344           Basic   
...       ...  ...     ...            ...             ...   
3995     3996   40    Male         127601         Premium   
3996     3997   40  Female          77944         Premium   
3997     3998   62    Male          33154           Basic   
3998     3999   55  Female         139822         Premium   
3999     4000   52  Female          61604           Basic   

      Membership Duration  Frequency of Visit Preferred Time  \
0                      46                   2      Afternoon   
1                      26                   6      Afternoon   
2                      46                  

#checking the type of data inside the data frame

In [5]:
df.dtypes


User ID                   int64
Age                       int64
Gender                   object
Annual Income             int64
Membership Type          object
Membership Duration       int64
Frequency of Visit        int64
Preferred Time           object
Previous Membership      object
Marital Status           object
Location                 object
Promotion Response       object
Health Condition         object
Membership Prediction    object
Professional             object
Professional Status      object
dtype: object

In [33]:
df

User ID  Age  Gender  Annual Income Membership Type  \
0           1   62   Other         135305         Premium   
1           2   65    Male          59056           Basic   
2           3   18    Male          63857           Basic   
3           4   21    Male         135072           Basic   
4           5   21  Female          59344           Basic   
...       ...  ...     ...            ...             ...   
3995     3996   40    Male         127601         Premium   
3996     3997   40  Female          77944         Premium   
3997     3998   62    Male          33154           Basic   
3998     3999   55  Female         139822         Premium   
3999     4000   52  Female          61604           Basic   

      Membership Duration  Frequency of Visit Preferred Time  \
0                      46                   2      Afternoon   
1                      26                   6      Afternoon   
2                      46                   2      Afternoon   
3                      54                   1        Evening   
4                      36                   5        Morning   
...                   ...                 ...            ...   
3995                   15                   1      Afternoon   
3996                   34                   1        Evening   
3997                   40                   1        Evening   
3998                   39                   3      Afternoon   
3999                    6                   6        Morning   

     Previous Membership Marital Status  Location Promotion Response  \
0                    Yes        Married     Rural                Yes   
1                    Yes         Single  Suburban                Yes   
2                     No        Married     Rural                Yes   
3                     No        Married  Suburban                 No   
4                    Yes         Single  Suburban                 No   
...                  ...            ...       ...                ...   
3995                 Yes         Single     Urban                 No   
3996                  No         Single     Urban                 No   
3997                 Yes        Married  Suburban                 No   
3998                 Yes         Single     Urban                 No   
3999                 Yes         Single  Suburban                Yes   

     Health Condition Membership Prediction Professional   Professional Status  
0             Average                    No  Late Career               Retired  
1                Good                    No      Retired               Retired  
2           Excellent                    No      Student               Student  
3                Good                   Yes      Student               Student  
4           Excellent                   Yes      Student               Student  
...               ...                   ...          ...                   ...  
3995             Good                    No   Mid Career  Working Professional  
3996          Average                    No   Mid Career  Working Professional  
3997             Good                   Yes  Late Career               Retired  
3998             Poor                   Yes  Late Career  Working Professional  
3999          Average                   Yes  Late Career  Working Professional  

[4000 rows x 16 columns]

#First 15 Rows to see what it looks like approximately

In [10]:
df.head(15)

User ID  Age  Gender  Annual Income Membership Type  Membership Duration  \
0         1   62   Other         135305         Premium                   46   
1         2   65    Male          59056           Basic                   26   
2         3   18    Male          63857           Basic                   46   
3         4   21    Male         135072           Basic                   54   
4         5   21  Female          59344           Basic                   36   
5         6   57    Male         130879         Premium                   49   
6         7   27  Female          99861             VIP                   12   
7         8   37  Female         136521             VIP                   14   
8         9   39  Female          36287         Premium                   15   
9        10   68    Male          66916           Basic                   36   
10       11   54  Female          87173         Premium                   56   
11       12   41  Female         145852           Basic                   25   
12       13   24  Female          26290           Basic                   37   
13       14   42  Female          72063           Basic                   24   
14       15   42    Male         133279           Basic                   58   

    Frequency of Visit Preferred Time Previous Membership Marital Status  \
0                    2      Afternoon                 Yes        Married   
1                    6      Afternoon                 Yes         Single   
2                    2      Afternoon                  No        Married   
3                    1        Evening                  No        Married   
4                    5        Morning                 Yes         Single   
5                    2      Afternoon                  No        Married   
6                    5        Morning                 Yes         Single   
7                    5        Evening                  No         Single   
8                    6        Evening                  No         Single   
9                    4        Morning                 Yes         Single   
10                   4        Evening                 Yes         Single   
11                   4      Afternoon                  No         Single   
12                   2        Morning                  No        Married   
13                   1        Evening                  No         Single   
14                   5        Morning                  No        Married   

    Location Promotion Response Health Condition Membership Prediction  \
0      Rural                Yes          Average                    No   
1   Suburban                Yes             Good                    No   
2      Rural                Yes        Excellent                    No   
3   Suburban                 No             Good                   Yes   
4   Suburban                 No        Excellent                   Yes   
5   Suburban                 No             Good                   Yes   
6   Suburban                Yes             Good                    No   
7      Urban                 No             Good                   Yes   
8      Urban                Yes             Good                   Yes   
9      Urban                 No             Good                    No   
10  Suburban                 No             Good                    No   
11     Urban                 No             Good                   Yes   
12     Rural                 No          Average                    No   
13     Urban                 No          Average                    No   
14     Rural                 No             Good                   Yes   

    Professional   Professional Status  
0    Late Career               Retired  
1        Retired               Retired  
2        Student               Student  
3        Student               Student  
4        Student               Student  
5    Late Career  Working Professional  
6   Early Career  Working Professi

#Statistics about the data frame

In [23]:
df.describe(include='all')

User ID          Age Gender  Annual Income Membership Type  \
count   4000.000000  4000.000000   4000    4000.000000            4000   
unique          NaN          NaN      3            NaN               3   
top             NaN          NaN   Male            NaN           Basic   
freq            NaN          NaN   1840            NaN            2434   
mean    2000.500000    43.633000    NaN   86346.917000             NaN   
std     1154.844867    15.182669    NaN   38029.388846             NaN   
min        1.000000    18.000000    NaN   20046.000000             NaN   
25%     1000.750000    30.000000    NaN   53324.250000             NaN   
50%     2000.500000    44.000000    NaN   86908.000000             NaN   
75%     3000.250000    57.000000    NaN  120324.000000             NaN   
max     4000.000000    69.000000    NaN  149980.000000             NaN   

        Membership Duration  Frequency of Visit Preferred Time  \
count           4000.000000         4000.000000           4000   
unique                  NaN                 NaN              3   
top                     NaN                 NaN        Morning   
freq                    NaN                 NaN           1595   
mean              30.301500            3.500500            NaN   
std               16.832407            1.709307            NaN   
min                1.000000            1.000000            NaN   
25%               16.000000            2.000000            NaN   
50%               31.000000            3.000000            NaN   
75%               45.000000            5.000000            NaN   
max               59.000000            6.000000            NaN   

       Previous Membership Marital Status Location Promotion Response  \
count                 4000           4000     4000               4000   
unique                   2              4        3                  2   
top                     No         Single    Urban                 No   
freq                  2447           1996     2049               2733   
mean                   NaN            NaN      NaN                NaN   
std                    NaN            NaN      NaN                NaN   
min                    NaN            NaN      NaN                NaN   
25%                    NaN            NaN      NaN                NaN   
50%                    NaN            NaN      NaN                NaN   
75%                    NaN            NaN      NaN                NaN   
max                    NaN            NaN      NaN                NaN   

       Health Condition Membership Prediction Professional  \
count              4000                  4000         4000   
unique                4                     2            5   
top                Good                   Yes  Late Career   
freq               1998                  2016         1141   
mean                NaN                   NaN          NaN   
std                 NaN                   NaN          NaN   
min                 NaN                   NaN          NaN   
25%                 NaN                   NaN          NaN   
50%                 NaN                   NaN          NaN   
75%                 NaN                   NaN          NaN   
max                 NaN                   NaN          NaN   

         Professional Status  
count                   4000  
unique                     3  
top     Working Professional  
freq                    2719  
mean                     NaN  
std                      NaN  
min                      NaN  
25%                      NaN  
50%                      NaN  
75%                      NaN  
max                      NaN

#Sanity Check, calculation of zero values column wise

In [19]:
df.isnull().sum()

User ID                  0
Age                      0
Gender                   0
Annual Income            0
Membership Type          0
Membership Duration      0
Frequency of Visit       0
Preferred Time           0
Previous Membership      0
Marital Status           0
Location                 0
Promotion Response       0
Health Condition         0
Membership Prediction    0
Professional             0
Professional Status      0
dtype: int64

#Sanity Check, calculation of zero values row wise

In [22]:
df.isnull().sum(axis=1)

0       0
1       0
2       0
3       0
4       0
       ..
3995    0
3996    0
3997    0
3998    0
3999    0
Length: 4000, dtype: int64

#unique values in column Professional Status

In [24]:
df['Professional Status'].unique()

array(['Retired', 'Student', 'Working Professional'], dtype=object)

In [25]:
df['Professional Status'].value_counts()

Professional Status
Working Professional    2719
Retired                  726
Student                  555
Name: count, dtype: int64

In [26]:
df['Professional Status']

0                    Retired
1                    Retired
2                    Student
3                    Student
4                    Student
                ...         
3995    Working Professional
3996    Working Professional
3997                 Retired
3998    Working Professional
3999    Working Professional
Name: Professional Status, Length: 4000, dtype: object

In [29]:
df['Professional Status'].isnull().max()

np.False_

#just checking manually, whether there are any 0 values

In [34]:
pd.set_option('display.max_rows', None)

In [ ]:
display(df)

#trying to find the rows with Single Retired People, 'srp_df'

In [42]:
options = ['Retired'] 
 

srp_df = df.loc[(df['Marital Status'] == 'Single') & 
              df['Professional Status'].isin(options)] 
srp_df.head(15)

User ID  Age  Gender  Annual Income Membership Type  Membership Duration  \
1          2   65    Male          59056           Basic                   26   
9         10   68    Male          66916           Basic                   36   
20        21   64    Male          92226         Premium                   35   
29        30   69  Female         104543           Basic                    1   
31        32   69    Male         120896         Premium                   19   
40        41   69  Female         141771         Premium                   56   
104      105   67    Male          94931             VIP                   48   
114      115   68  Female          51272         Premium                   30   
123      124   66  Female         148494         Premium                   11   
130      131   61    Male          53152           Basic                    3   
138      139   64    Male         109609           Basic                   10   
154      155   69    Male          98226           Basic                   14   
171      172   61  Female          68851           Basic                   47   
187      188   63   Other          29600           Basic                   26   
218      219   65    Male         106048         Premium                   57   

     Frequency of Visit Preferred Time Previous Membership Marital Status  \
1                     6      Afternoon                 Yes         Single   
9                     4        Morning                 Yes         Single   
20                    6        Morning                 Yes         Single   
29                    2        Morning                  No         Single   
31                    4        Morning                  No         Single   
40                    1        Evening                  No         Single   
104                   2        Morning                  No         Single   
114                   4      Afternoon                 Yes         Single   
123                   2      Afternoon                 Yes         Single   
130                   3        Morning                  No         Single   
138                   6      Afternoon                 Yes         Single   
154                   4      Afternoon                 Yes         Single   
171                   6      Afternoon                  No         Single   
187                   3        Morning                  No         Single   
218                   3        Morning                  No         Single   

     Location Promotion Response Health Condition Membership Prediction  \
1    Suburban                Yes             Good                    No   
9       Urban                 No             Good                    No   
20      Urban                Yes        Excellent                    No   
29      Rural                 No        Excellent                   Yes   
31   Suburban                 No        Excellent                    No   
40      Urban                 No          Average                   Yes   
104  Suburban                 No             Poor                    No   
114     Urban                 No        Excellent                    No   
123  Suburban                Yes             Good                    No   
130     Urban                 No        Excellent                   Yes   
138     Rural                 No             Good                   Yes   
154     Rural                Yes             Poor                   Yes   
171     Rural                 No        Excellent                   Yes   
187     Urban                 No             Good                    No   
218     Rural                 No        Excellent                   Yes   

    Professional Professional Status  
1        Retired             Retired  
9        Retired             Retired  
20   Late Career             Retired  
29       Retired             Retired  
31       Retired             Retired  
40       Retired             Retired  
1

#So now, that I have found all the single retired people, lets calculate, what is the count of men and women, who are retired and single 

In [44]:
options = ['Retired'] 
 

srp_df = df.loc[(df['Marital Status'] == 'Single') & df['Professional Status'].isin(options)] 
srp_df['Gender'].value_counts()

Gender
Male      163
Female    157
Other      20
Name: count, dtype: int64

#So now we know, that we have 163 retired men and 157 retired women, who are going to the gym, lets see, who is going in the morning, and who is going in the afternoon

In [46]:
options = ['Retired'] 
srp_df = df.loc[(df['Marital Status'] == 'Single') & df['Professional Status'].isin(options)] 
srp_df['Preferred Time'].value_counts()

Preferred Time
Morning      146
Afternoon     99
Evening       95
Name: count, dtype: int64

Now we see who is joining at what time, lets see, who is joining morning more often, women or men

This is for Male

In [47]:
options = ['Retired'] 
srp_df = df.loc[(df['Marital Status'] == 'Single') &(df['Gender']== 'Male') & df['Professional Status'].isin(options)] 
srp_df['Preferred Time'].value_counts()

Preferred Time
Morning      72
Afternoon    50
Evening      41
Name: count, dtype: int64

This is for the Women

In [48]:
options = ['Retired'] 
srp_df = df.loc[(df['Marital Status'] == 'Single') &(df['Gender']== 'Female') & df['Professional Status'].isin(options)] 
srp_df['Preferred Time'].value_counts()

Preferred Time
Morning      62
Evening      49
Afternoon    46
Name: count, dtype: int64

And for the other gender types, that are provided in the data frame

In [49]:
options = ['Retired'] 
srp_df = df.loc[(df['Marital Status'] == 'Single') &(df['Gender']== 'Other') & df['Professional Status'].isin(options)] 
srp_df['Preferred Time'].value_counts()

Preferred Time
Morning      12
Evening       5
Afternoon     3
Name: count, dtype: int64

Now lets see, who is going to the gym more often, male, female or other

In [62]:
options = ['Retired'] 
srp_df = df.loc[(df['Marital Status'] == 'Single') &(df['Gender']== 'Other') & df['Professional Status'].isin(options)] 
print('Other Gender are going in average',srp_df['Frequency of Visit'].value_counts().mean().round(),'times')
srp_df = df.loc[(df['Marital Status'] == 'Single') &(df['Gender']== 'Male') & df['Professional Status'].isin(options)] 
print('Men are going in average',srp_df['Frequency of Visit'].value_counts().mean().round(),'times')
srp_df = df.loc[(df['Marital Status'] == 'Single') &(df['Gender']== 'Female') & df['Professional Status'].isin(options)] 
print('Women are going in average',srp_df['Frequency of Visit'].value_counts().mean().round(),'times')

Other Gender are going in average 4.0 times
Men are going in average 27.0 times
Women are going in average 26.0 times


#Now I want to see, out our specifies category, what is the average income 


In [63]:
df.columns

Index(['User ID', 'Age', 'Gender', 'Annual Income', 'Membership Type',
       'Membership Duration', 'Frequency of Visit', 'Preferred Time',
       'Previous Membership', 'Marital Status', 'Location',
       'Promotion Response', 'Health Condition', 'Membership Prediction',
       'Professional', 'Professional Status'],
      dtype='object')

In [116]:
Retired_df = df[(df['Marital Status'] == 'Single') & (df['Professional Status'] == 'Retired')]
Retired_df.head(5)

User ID  Age  Gender  Annual Income Membership Type  Membership Duration  \
1         2   65    Male          59056           Basic                   26   
9        10   68    Male          66916           Basic                   36   
20       21   64    Male          92226         Premium                   35   
29       30   69  Female         104543           Basic                    1   
31       32   69    Male         120896         Premium                   19   

    Frequency of Visit Preferred Time Previous Membership Marital Status  \
1                    6      Afternoon                 Yes         Single   
9                    4        Morning                 Yes         Single   
20                   6        Morning                 Yes         Single   
29                   2        Morning                  No         Single   
31                   4        Morning                  No         Single   

    Location Promotion Response Health Condition Membership Prediction  \
1   Suburban                Yes             Good                    No   
9      Urban                 No             Good                    No   
20     Urban                Yes        Excellent                    No   
29     Rural                 No        Excellent                   Yes   
31  Suburban                 No        Excellent                    No   

   Professional Professional Status  
1       Retired             Retired  
9       Retired             Retired  
20  Late Career             Retired  
29      Retired             Retired  
31      Retired             Retired

So now we see the average income of single retired gym members by gender

In [134]:
group_income = Retired_df.groupby('Gender').agg({'Annual Income': ['mean']})
group_visit = Retired_df.groupby('Gender').agg({'Frequency of Visit': ['mean']})
group_membership_duration = Retired_df.groupby('Gender').agg({'Membership Duration': ['mean']})
group_visit
group_income
group_membership_duration
gro




Membership Duration
                      mean
Gender                    
Female           29.866242
Male             30.588957
Other            28.550000